STL_Extension/include/CGAL/Pair_optional_adaptor.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2024 GeometryFactory Sarl (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: LGPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     : Sven Oesau
+
+#ifndef CGAL_PAIR_OPTIONAL_ADAPTOR
+#define CGAL_PAIR_OPTIONAL_ADAPTOR
+
+namespace CGAL {
+
+// T is supposed to be a handle
+template<typename T>
+class Pair_optional_adaptor : public std::optional<T> {
```

Comment:
I'm not sure you need to inherit from optional. I think I'd simply go for storing first and second and have conversion operators. Remember that both first and second will be lightweigth to you should not bother with references, union, ...

Reply:
And I don't now how to really do without references. The type of first is T, if the optional I get is nullopt, I still need to provide an instance of T. It can be invalid, though. Thus, I would need to require T to be default constructible or use a union to avoid that.

Reply:
It is already default constructible since that's how it is currently implemented using a pair.

Reply:
It is fully working now. I can still switch to simply storing first and second if you prefer or think it looks cleaner.